compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -5193,8 +5203,32 @@ private BSymbol getLangLibMethod(BLangInvocation iExpr, BType bType) {
         return funcSymbol;
     }
 
+    private BType createXMLNeverType(BLangInvocation iExpr, BLangXMLTextLiteral bLangXMLTextLiteral) {
```

Comment:
Instead of creating a new type, we can have a `xmlNeverType` in the symbol table and reuse it.